src/Elasticsearch/State/CollectionProvider.php

```diff
@@ -60,13 +59,20 @@ public function provide(Operation $operation, array $uriVariables = [], array $c
         $limit = $body['size'] ??= $this->pagination->getLimit($operation, $context);
         $offset = $body['from'] ??= $this->pagination->getOffset($operation, $context);
 
+        $options = $operation->getStateOptions();
+        if (null === $options) {
+            $options = $this->convertDocumentMetadata($this->documentMetadataFactory->create($resourceClass));
+        } elseif (!$options instanceof OptionsInterface) {
+            throw new \LogicException(sprintf('Operation‘s state options must be null or an instance of %s, but got %s', OptionsInterface::class, \get_class($options)));
+        }
+
         $params = [
-            'index' => $documentMetadata->getIndex(),
+            'index' => $options->getIndex() ?? Inflector::tableize($operation->getShortName()),
```

Comment:
see above

Reply:
I think we should keep this because `$index` in `ApiPlatform/Elasticsearch/State/Options` can be null. But is it really necessary to keep `$index` nullable ?